lib/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { createEvaluator, FlagCallbacks } from 'flagr-feature-typescript';
+
+type FlagrContextType = {
+  evaluate?: <T>(flag: string, callbacks: FlagCallbacks<T>) => T;
+  match?: (flag: string, matchVariant?: string) => boolean;
+};
+
+const FlagrContext = createContext<FlagrContextType>({});
+
+export const useFlagr = () => useContext(FlagrContext);
+
+interface FlagrContextProviderProps {
+  children: React.ReactNode;
+  flagrUrl: string;
+  tags: [string, ...string[]];
+}
+
+export const FlagrContextProvider = ({
+  children,
+  flagrUrl,
+  tags,
+}: FlagrContextProviderProps) => {
+  const [value, setValue] = useState<FlagrContextType>({});
+
+  useEffect(() => {
+    (async () => {
+      const Evaluator = createEvaluator({
+        flagrUrl,
+      });
+
+      const { cachedEvaluate, cachedMatch } = await Evaluator.batchEvaluation({
+        context: {},
```

Comment:
**thought:** we're going to want to be able to specify this in the not too distant future. Non-plussed as to whether it's promoted to prop now or laterz

Reply:
Added now